manifests/repo/apt_postgresql_org.pp

```diff
@@ -11,7 +11,11 @@
   #
   $default_baseurl = 'http://apt.postgresql.org/pub/repos/apt/'
 
-  $_baseurl = pick($postgresql::repo::baseurl, $default_baseurl)
+  if $lsbdistcodename == 'stretch' {
+    $_baseurl = 'https://apt-archive.postgresql.org/pub/repos/apt'
```

Comment:
Also this would break any usage of the hiera setup in case the distro is `stretch`, while using it on any other server. That alone is a good reason not to merge this.